app/src/main/java/com/sapuseven/untis/ui/activities/InfoCenter.kt

```diff
@@ -438,3 +556,196 @@ private fun formatOfficeHourTime(
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun AbsenceFilterDialog(
+	initialOrder: AbsenceOrder,
+	onDismiss: (requestedAbsenceOrder: AbsenceOrder) -> Unit,
+) {
+	var absenceOrder by rememberSaveable(
+		saver = Saver(
+			save = { Json.encodeToString(it.value) },
+			restore = { mutableStateOf(Json.decodeFromString(it)) }
+		)
+	) { mutableStateOf(initialOrder) }
+	var dismissed by rememberSaveable { mutableStateOf(false) }
+
+
+	fun dismiss(absenceOrder: AbsenceOrder) {
+		onDismiss(absenceOrder)
+		dismissed = true
+	}
+
+	BackHandler(
+		enabled = !dismissed,
+	) {
+		dismiss(absenceOrder)
+	}
+
+	Scaffold(
+		topBar = {
+			CenterAlignedTopAppBar(
+				title = { Text(text = "Filter Absencses") },
+				navigationIcon = {
+					IconButton(onClick = {
+						dismiss(absenceOrder)
+					}) {
+						Icon(
+							imageVector = Icons.Outlined.Close,
+							contentDescription = stringResource(id = R.string.all_close)
+						)
+					}
+				}
+			)
+		}
+	) { innerPadding ->
+		VerticalScrollColumn(paddingValues = innerPadding) {
+			PreferenceCategory(title = "") {
+				ListItem(
+					headlineText = {
+						Box {
+							Text(text = "Show unexcused only")
+						}
+					},
+					trailingContent = {
+						Switch(
+							checked = absenceOrder.orderType.showOnlyUnexcused,
+							onCheckedChange = { newValue ->
+								absenceOrder = absenceOrder.copy(absenceOrder.orderType.copy(newValue))
+							}
+						)
+					},
+				)
+				ListItem(
+					headlineText = {
+						Box {
+							Text(text = "Sort by")
+						}
+					},
+					supportingText = {
+						if (absenceOrder.orderType is OrderType.Ascending) {
+							Text(text = "Ascending")
+						} else {
+							Text(text = "Descending")
+						}
+					},
+					trailingContent = {
+						Switch(
+							checked = absenceOrder.orderType is OrderType.Ascending,
+							onCheckedChange = { ascending ->
+								absenceOrder = if (ascending) {
+									absenceOrder.copy(OrderType.Ascending(absenceOrder.orderType.showOnlyUnexcused))
+								} else {
+									absenceOrder.copy(OrderType.Descending(absenceOrder.orderType.showOnlyUnexcused))
+								}
+
+							}
+						)
+					},
+				)
+				ListSelector(
+					title = { Text(text = "Time Ranges") },
+					absenceOrder = absenceOrder,
+					entries = arrayOf(
+						AbsenceOrder.LastSevenDays(absenceOrder.orderType),
+						AbsenceOrder.LastFourteenDays(absenceOrder.orderType),
+						AbsenceOrder.LastThirtyDays(absenceOrder.orderType),
+						AbsenceOrder.LastNinetyDays(absenceOrder.orderType),
+						AbsenceOrder.CurrentSchoolYear(absenceOrder.orderType),
+					),
+				) {
+					absenceOrder = it
+				}
+			}
+		}
+	}
+}
+
+@Composable
+private fun VerticalScrollColumn(paddingValues: PaddingValues, content: @Composable ColumnScope.() -> Unit) {
+	Column(
+		horizontalAlignment = Alignment.CenterHorizontally,
+		modifier = Modifier
+			.verticalScroll(rememberScrollState())
+			.bottomInsets()
+			.padding(paddingValues)
```

Comment:
Is it really necessary to use `.padding()` when `.bottomInsets()` already adds required paddings?
I believe it should be enough to use the `VerticalScrollColumn` from `SettingsActivity` - maybe extract it to `ui.common.ScrollColumn` (or something like that)